### PR TITLE
[runtime-security] Fix dentry inode when accessing lower layer

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR fixes the inode reported when accessing a file that is present from a lower layer.

### Motivation

The inode was incorrect thus the dentry resolution was wrong.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Added functional tests

### CODE REVIEW

File: pkg/security/tests/dentry_test.go [hunk 1]
```diff
@@ -310,3 +311,109 @@ func TestDentryRmdir(t *testing.T) {
...
+		Expression: `open.filename == "{{.Root}}/merged/kiki.txt"`,
+	}
+
+	test, err := newTestModule(nil, []*rules.RuleDefinition{rule}, testOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	testLower, testUpper, testWordir, testMerged := createOverlayLayers(t, test)
```
Comment (guedou) [inline pkg/security/tests/dentry_test.go:348]: nit: delete these directories at the end of the test.

**Review by guedou — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
LGTM. It also fixes the "alert on second trigger" feature in containers 🎉

File: pkg/security/tests/dentry_test.go (hunk 1 above)
Comment (safchain) [reply pkg/security/tests/dentry_test.go:348]: I think this is done by this line
https://github.com/DataDog/datadog-agent/blob/master/pkg/security/tests/setup_test.go#L398 ?

Comment (guedou) [reply pkg/security/tests/dentry_test.go:348]: Thanks.

**Pull request merged**